From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Thu, 3 Dec 2020 20:24:53 +0100
Subject: [PATCH] fdts: stm32mp1: enable hash1 unconditionally

&hash1 is an internal IP that can be enabled always as it has no
dependencies. Indeed we must enable it, because TF-A panics otherwise
during stm32mp_init_auth(). Instead of duplicating the enabling
line in every board device tree, stick it into the SoC dtsi.

This likely fixes boot on the Avenger board that lacks the status =
"okay" line for the &hash1 device tree node.

Change-Id: I5901029f29af0cf213510878b5daaf5240f96f17
Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 fdts/stm32mp151.dtsi      | 2 +-
 fdts/stm32mp157c-ed1.dts  | 4 ----
 fdts/stm32mp15xx-dkx.dtsi | 4 ----
 3 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/fdts/stm32mp151.dtsi b/fdts/stm32mp151.dtsi
index 8f175a6492a1..d908b710bbb9 100644
--- a/fdts/stm32mp151.dtsi
+++ b/fdts/stm32mp151.dtsi
@@ -253,7 +253,7 @@
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc HASH1>;
 			resets = <&rcc HASH1_R>;
-			status = "disabled";
+			status = "okay";
 		};
 
 		rng1: rng@54003000 {
diff --git a/fdts/stm32mp157c-ed1.dts b/fdts/stm32mp157c-ed1.dts
index a6b98b7d9337..615e2cc07578 100644
--- a/fdts/stm32mp157c-ed1.dts
+++ b/fdts/stm32mp157c-ed1.dts
@@ -55,10 +55,6 @@
 	status="okay";
 };
 
-&hash1 {
-	status = "okay";
-};
-
 &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4_pins_a>;
diff --git a/fdts/stm32mp15xx-dkx.dtsi b/fdts/stm32mp15xx-dkx.dtsi
index 52b914b84e45..e59a65fb5555 100644
--- a/fdts/stm32mp15xx-dkx.dtsi
+++ b/fdts/stm32mp15xx-dkx.dtsi
@@ -41,10 +41,6 @@
 	cpu-supply = <&vddcore>;
 };
 
-&hash1 {
-	status = "okay";
-};
-
 &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4_pins_a>;
